oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/ThrottlingStatsCollectorImpl.java

```diff
@@ -126,12 +126,22 @@ public void doneCreateOrUpdate(long throttlingTimeNanos, Collection<? extends Do
     public void doneFindAndModify(long throttlingTimeNanos, Collection<? extends Document> collection, String key, boolean newEntry,
                                   boolean success, int retryCount) {
 
-        modifyMetricUpdater.update(collection, retryCount, throttlingTimeNanos, success, newEntry, c -> c == NODES,
+        modifyMetricUpdater.update(collection, retryCount, throttlingTimeNanos, success, newEntry, of(key), isNodesCollectionUpdated(),
                 getStatsConsumer(), getStatsConsumer(), MeterStats::mark, MeterStats::mark);
 
         perfLog(perfLog, PERF_LOG_THRESHOLD, throttlingTimeNanos, "findAndModify [{}]", key);
     }
 
+    @Override
+    public void doneFindAndModify(long throttlingTimeNanos, Collection<? extends Document> collection, List<String> ids, boolean newEntry,
+                                  boolean success, int retryCount) {
+
+        modifyMetricUpdater.update(collection, retryCount, throttlingTimeNanos, success, newEntry, ids, isNodesCollectionUpdated(),
+                getStatsConsumer(), getStatsConsumer(), MeterStats::mark, MeterStats::mark);
+
+        perfLog(perfLog, PERF_LOG_THRESHOLD, throttlingTimeNanos, "findAndModify [{}]", ids);
```

Comment:
Remove square brackets?
```suggestion
        perfLog(perfLog, PERF_LOG_THRESHOLD, throttlingTimeNanos, "findAndModify {}", ids);
```